js/src/webrtc.js

```diff
@@ -4,36 +4,42 @@ require('webrtc-adapter');
 import * as html2canvas from 'html2canvas';
 import * as mqtt from 'mqtt';
 import * as utils from './utils';
-var semver_range = '~' + require('../package.json').version;
+const semver_range = '~' + require('../package.json').version;
 
-var MediaStreamModel = widgets.DOMWidgetModel.extend({
-    defaults: function() {
-        return _.extend(widgets.DOMWidgetModel.prototype.defaults(), {
+import { imageWidgetToCanvas } from './utils';
+
+export class MediaStreamModel extends widgets.DOMWidgetModel {
+    defaults() {
+        return {...super.defaults(),
             _model_module: 'jupyter-webrtc',
             _view_module: 'jupyter-webrtc',
             _model_name: 'MediaStreamModel',
             _view_name: 'MediaStreamView',
             _model_module_version: semver_range,
             _view_module_version: semver_range,
-        });
-    },
-});
-
-// Backward compatibility
-Object.defineProperty(MediaStreamModel.prototype, 'stream', {
-    get: function() { return this.captureStream(); }
-});
-
-var captureStream = function(widget) {
-    if(widget.captureStream)
-        return widget.captureStream()
-    else
-        return widget.stream
+        };
+    }
+
+    get stream() {
+        return this.captureStream();
+    }
+
+    captureStream() {
+        throw new Error('Not implemented');
+    }
```

Comment:
I adder this just to indicate that this is an abstract base class, where this method should be overloaded.